torch/csrc/jit/script/module.h

```diff
@@ -762,6 +742,16 @@ struct Module {
     return list.back();
   }
 
+  // add a new entry to the singleton object that represents this
+  // Module as a first-class value in code, and update the corresponding
+  // ClassType to match.
+  Slot appendSlot(const std::string& name, TypePtr typ, IValue value) {
+    const ClassTypePtr& type = module_value_->type();
+    type->addAttribute(name, std::move(typ));
+    module_value_->appendSlot(std::move(value));
+    return Slot(module_value_, type->attributeNames().size() - 1);
```

Comment:
I think use `type->getAttributeSlot` here, as it will be more robust if we ever change the storage of attributes on types (the current code assumes that adding to an attribute will append to a vector).